internal/controller/progressive/progressive.go

```diff
@@ -278,8 +278,14 @@ func processUpgradingChild(
 	}
 	childStatus = rolloutObject.GetUpgradingChildStatus()
 
-	// check for Force Promote set in Progressive strategy to force success logic
-	if rolloutObject.GetProgressiveStrategy().ForcePromote {
+	forcePromote := false
+	_, ok := existingUpgradingChildDef.GetLabels()[common.LabelKeyForcePromote]
+	if ok {
+		forcePromote = true
+	}
+
+	// check for Force Promote set in Progressive strategy to force success logic OR if promoted child has force-promote label
```

Comment:
why does it say "if promoted child" rather than "if upgrading child"? did you test it with promoted child or upgrading child?